Source/WebCore/page/playstation/ResourceUsageThreadPlayStation.cpp

```diff
@@ -55,17 +57,40 @@ void ResourceUsageThread::platformCollectCPUData(JSC::VM*, ResourceUsageData& da
 
 void ResourceUsageThread::platformCollectMemoryData(JSC::VM* vm, ResourceUsageData& data)
 {
-    // FIXME: Calculate total dirty memory size for ResourceUsageOverlay
-    data.totalDirtySize = 0;
+    auto& categories = data.categories;
+
+    data.totalDirtySize = bmalloc::api::memoryFootprint();
 
     size_t currentGCHeapCapacity = vm->heap.blockBytesAllocated();
     size_t currentGCOwnedExtra = vm->heap.extraMemorySize();
     size_t currentGCOwnedExternal = vm->heap.externalMemorySize();
     RELEASE_ASSERT(currentGCOwnedExternal <= currentGCOwnedExtra);
 
-    data.categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
-    data.categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
-    data.categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+    categories[MemoryCategory::GCHeap].dirtySize = currentGCHeapCapacity;
+    categories[MemoryCategory::GCOwned].dirtySize = currentGCOwnedExtra - currentGCOwnedExternal;
+    categories[MemoryCategory::GCOwned].externalSize = currentGCOwnedExternal;
+
+    if (isFastMallocEnabled()) {
+        auto& mallocBucket = categories[MemoryCategory::bmalloc];
+
+        showmap::Result<256> result;
+        result.collect();
+        if (auto* entry = result.entry("SceNKFastMalloc"))
+            mallocBucket.dirtySize = entry->effectiveRss();
+
+        // TODO: collect dirty size of "MemoryCategory::Images" and "MemoryCategory::Layers"
+        if (data.totalDirtySize > mallocBucket.dirtySize)
+            categories[MemoryCategory::Other].dirtySize = data.totalDirtySize - mallocBucket.dirtySize;
+
+        mallocBucket.dirtySize -= currentGCHeapCapacity;
+        mallocBucket.dirtySize -= std::min(mallocBucket.dirtySize, categories[MemoryCategory::GCOwned].dirtySize);
+    } else {
+        auto& mallocBucket = categories[MemoryCategory::Other];
+        mallocBucket.dirtySize = data.totalDirtySize;
+
+        mallocBucket.dirtySize -= currentGCHeapCapacity;
+        mallocBucket.dirtySize -= std::min(mallocBucket.dirtySize, categories[MemoryCategory::GCOwned].dirtySize);
```

Comment:
Else part of this four lines can be written in one line:
```
categories[MemoryCategory::Other] = data.totalDirtySize - std::min(data.totalDirtySize, currentGCHeapCapacity + currentGCOwnedExtra - currentGCOwnedExternal);
```
or two lines for readability:
```
auto currentGCDirtySize = categories[MemoryCategory::GCHeap].dirtySize + categories[MemoryCategory::GCOwned].dirtySize;
categories[MemoryCategory::Other] = data.totalDirtySize - std::min(data.totalDirtySize, currentGCDirtySize);
```

Reply:
And definition of `currentGCDirtySize`  can be moved just after definition of `currentGCOwnedExternal` using original values:
```
auto currentGCDirtySize = currentGCHeapCapacity + currentGCOwnedExtra - currentGCOwnedExternal;
```
so the else part is again one line :)